File: front/src/js/reducers/step.js
import * as type from '../actions/const'

const initialStep = {
  id: 0,
  repetitions: 8,
  series: 3,
  weight: 0,
  rest: 45,
  rest_end: 90,
  exercise: {
    image: ""
  }
}

export function currentStep(state=initialStep, action){
  switch (action.type) {
    case type.CURRENT_STEP:
      return action.payload.step
    case type.INITIAL_CURRENT_STEP:
      return initialStep
    default:
      return state
  }
}

export function editStep(state=-1, action){
  switch (action.type) {
    case type.ID_STEP:
      return action.payload.id
    case type.PROGRAMS:
    case type.PROGRAM:
    case type.EXERCISES:
      return -1
    default:
      return state
  }
}

export function steps(state=[], action){
  switch (action.type) {
    case type.CURRENT_PROGRAM:
      return action.payload.program.steps
    case type.CHANGE_POSITION:
      return action.payload.steps.map((step, index) => {
        step.position = index+1
        return step
      })
    case type.ADD_STEP:
      return [...state, action.payload.step]
    case type.DELETE_STEP:
      return state.filter(step => step.id != action.payload.id)
        .map((step, index) => {
          step.position = index+1
          return step
        })
    case type.EDIT_STEP:
      return state.map(step => {
        if (step.id == action.payload.step.id) {
          step = action.payload.step
        }
        return step
      })
    default:
      return state
  }
}

File: back/sport_programs/models/step.py
from sport_programs import db, ma

class Step(db.Model):
    __tablename__ = 'step'

    id = db.Column(db.Integer, primary_key=True)
    program_id = db.Column(db.Integer, db.ForeignKey('program.id'), nullable=False)
    exercise_id = db.Column(db.Integer, db.ForeignKey('exercise.id'), nullable=False)
    series = db.Column(db.Integer, default=3, nullable=False)
    repetitions = db.Column(db.Integer, default=8, nullable=False)
    weight = db.Column(db.Integer, default=0, nullable=False)
    rest = db.Column(db.Integer, default=60, nullable=False)
    rest_end = db.Column(db.Integer, default=60, nullable=False)
    position = db.Column(db.Integer, nullable=False)
